repobee_sanitizer/_sanitize.py

```diff
@@ -62,3 +91,18 @@ def _sanitize(lines: List[str]) -> Iterable[str]:
             continue
         if keep:
             yield line[prefix_length:]
+
+
+def _strip(lines: List[str]) -> Iterable[str]:
+    keep = True
+    for line in lines:
+        marker = _syntax.contained_marker(line)
+        if marker == Markers.START or marker == Markers.SHRED:
+            continue
+        elif marker == Markers.REPLACE:
+            keep = False
+        elif marker == Markers.END:
+            keep = True
+            continue
+        if keep:
+            yield line
```

Comment:
This is a bit redundant, the strip functionality just requires us to reverse the handling of the blocks. Here's a way to implement it by extending the `_sanitizer` function with a new `strip` parameter.

```suggestion
def _sanitize(lines: List[str], strip: bool = False) -> Iterable[str]:
    keep = True
    prefix_length = 0
    for line in lines:
        marker = _syntax.contained_marker(line)
        if marker == Markers.START:
            prefix = re.match(rf"(.*?){Markers.START.value}", line).group(1)
            prefix_length = len(prefix) if not strip else 0
            keep = strip
        elif marker == Markers.REPLACE:
            keep = not strip
        elif marker == Markers.END:
            prefix_length = 0
            keep = True

        if keep and not marker:
            yield line[prefix_length:]
```